src/main/webapp/app/exam/manage/exams/exam-update.component.ts

```diff
@@ -25,118 +28,185 @@ export class ExamUpdateComponent implements OnInit {
+            const exam: Exam = isImport ? prepareExamForImport(data.exam) : data.exam;
 
-            // Tap the URL to determine, if the Exam should be imported
-            this.route.url.pipe(tap((segments) => (this.isImport = segments.some((segment) => segment.path === 'import')))).subscribe();
-
-            if (this.isImport) {
-                this.resetIdAndDatesForImport();
-                this.isImportInSameCourse = this.exam.course?.id === Number(this.route.snapshot.paramMap.get('courseId'));
+            if (!exam.gracePeriod) {
+                exam.gracePeriod = 180;
             }
 
-            this.courseManagementService.find(Number(this.route.snapshot.paramMap.get('courseId'))).subscribe({
-                next: (response: HttpResponse<Course>) => {
-                    this.exam.course = response.body!;
-                    this.course = response.body!;
-                    this.hideChannelNameInput = (exam.id !== undefined && exam.channelName === undefined) || !isMessagingEnabled(this.course);
-                },
-                error: (err: HttpErrorResponse) => onError(this.alertService, err),
-            });
-
-            if (!this.exam.gracePeriod) {
-                this.exam.gracePeriod = 180;
-            }
             // test exam only feature automatic assessment
-            if (this.exam.testExam) {
-                this.exam.numberOfCorrectionRoundsInExam = 0;
-            } else if (!this.exam.numberOfCorrectionRoundsInExam) {
-                this.exam.numberOfCorrectionRoundsInExam = 1;
+            if (exam.testExam) {
+                exam.numberOfCorrectionRoundsInExam = 0;
+            } else if (!exam.numberOfCorrectionRoundsInExam) {
+                exam.numberOfCorrectionRoundsInExam = 1;
             }
+
+            this.exam = exam;
+            this.isImport = isImport;
+            this.isImportInSameCourse = isImport && exam.course?.id === data.course.id;
+            this.originalStartDate = exam.startDate?.clone();
+            this.originalEndDate = exam.endDate?.clone();
+
+            this.course = data.course;
+            this.exam.course = data.course;
+            this.hideChannelNameInput = (!!exam.id && !exam.channelName) || !isMessagingEnabled(this.course);
         });
-        // Initialize helper attributes
-        this.workingTimeInMinutes = this.exam.workingTime! / 60;
-        this.calculateMaxWorkingTime();
+    }
+
+    /**
+     * Sets the exam working time in minutes.
+     * @param minutes
+     */
+    set workingTimeInMinutes(minutes: number) {
+        this.exam.workingTime = minutes * 60;
+    }
+
+    /**
+     * Returns the exam working time in minutes.
+     */
+    get workingTimeInMinutes(): number {
+        return this.exam.workingTime ? this.exam.workingTime / 60 : 0;
+    }
+
+    get oldWorkingTime(): number | undefined {
+        return normalWorkingTime({ startDate: this.originalStartDate, endDate: this.originalEndDate } as Exam);
```

Comment:
Can you change the method signature to `normalWorkingTime(startDatte, endDate)` to avoid this object creation + cast`?

Reply:
I've introduced an additional `examWorkingTime` helper, which allows passing the exam. The `normalWorkingTime` now allows to pass a `startDate` and `endDate`.